torch/csrc/jit/mobile/import.cpp

```diff
@@ -285,14 +418,24 @@ mobile::Module BytecodeDeserializer::deserialize(
   // being a Tuple (int, table), and the integer stands for the bytecode version
   // number. The rest of the elements are the same as before.
   //
+  auto meta_dict = readMobileMetadata(mcu);
   auto bvals = readArchive("bytecode", mcu).toTuple()->elements();
 
   c10::optional<std::vector<IValue>> debug_info_bvals;
   if (reader_->hasRecord("mobile_debug.pkl")) {
     debug_info_bvals = readArchive("mobile_debug", mcu).toTuple()->elements();
   }
-  parseMethods(bvals, debug_info_bvals, *mcu);
-  auto meta_dict = readMobileMetadata(mcu);
+
+  std::vector<IValue> constant_values_from_jit;
+  bool need_tensor_jit = has_tensor_jit_index(bvals);
+
+  // Read from constants.pkl only if there exists tensor jit index in
+  // bytecode.pkl
+  if (need_tensor_jit && reader_->hasRecord("constants.pkl")) {
```

Comment:
I agree with @dhruvbird. But I would also say why have `has_tensor_jit_index` at all? You can just do `constant_values_from_jit =  readArchive("constants", mcu).toTuple()->elements();` unconditionally? It probably will add to load time, but most likely this condition will be true for the models saved in future with this optimization. That way we also get less code?

Reply:
Would it bring more memory usage for the old models?